server/test/resolve.cpp

```diff
@@ -1211,3 +1211,194 @@ TEST_CASE("Resolve by hosted by")
         FAIL(ex.what());
     }
 }
+
+TEST_CASE("Resolve by Group")
+{
+    try {
+        fty::SampleDb db(R"(
+            items:
+                - type : Datacenter
+                  name : datacenter
+                  items:
+                    - type     : Server
+                      name     : srv11
+                      ext-name : srv11
+                - type : Datacenter
+                  name : datacenter1
+                  items:
+                    - type     : Server
+                      name     : srv21
+                      ext-name : srv21
+                - type : Datacenter
+                  name : datacenter2
+                  items:
+                    - type     : Server
+                      name     : serv
+                      ext-name : serv
+                    - type     : Server
+                      name     : servsrv
+                      ext-name : servsrv
+            )");
+
+        Group group;
+        group.name          = "ByName";
+        group.rules.groupOp = fty::Group::LogicalOp::And;
+
+        Group group1;
+        group1.name          = "EmptyWithLink";
+        group1.rules.groupOp = fty::Group::LogicalOp::And;
+
+        {
+            auto& groupMock = group.rules.conditions.append();
+            auto& condMock  = groupMock.reset<fty::Group::Condition>();
+            condMock.field  = fty::Group::Fields::Name;
+
+            condMock       = group.rules.conditions[0].get<fty::Group::Condition>();
+            condMock.value = "srv";
+            condMock.op    = fty::Group::ConditionOp::Contains;
+
+            auto& groupName = group1.rules.conditions.append();
+            auto& condName  = groupName.reset<fty::Group::Condition>();
+            condName.field  = fty::Group::Fields::Name;
+
+            auto& groupGroupLink = group1.rules.conditions.append();
+            auto& condGroupLink  = groupGroupLink.reset<fty::Group::Condition>();
+            condGroupLink.field  = fty::Group::Fields::Group;
+        }
+        auto g = group.create();
```

Comment:
just 1 latter g? Please, try to use more :)
If you look to the other tests, you will find other way how to create group, try it, maybe json string will be more descriptive for you.